### PR TITLE
Add a script that tests formatter stability on repositories

### PR DESCRIPTION
**Pull request by konstin**
## Summary

 We want to ensure that once formatted content stays the same when formatted again, which is known as formatter stability or formatter idempotency, and that the formatter prints syntactically valid code. As our test cases cover only a limited amount of code, this allows checking entire repositories.

 This adds a new subcommand to `ruff_dev` which can be invoked as `cargo run --bin ruff_dev -- check-formatter-stability <repo>`. While initially only intended to check stability, it has also found cases where the formatter printed invalid syntax or panicked.

 ## Test Plan

 Running this on cpython is already identifying bugs (https://github.com/astral-sh/ruff/pull/5089)

### CODE REVIEW

**Comment by konstin** [comment]
As someone who has little experience with fuzzer, what would it mean to use cpython for the initial corpus?

**Comment by addisoncrump** [comment]
The corpus is the set of inputs on which the fuzzer tries to modify such that it triggers new code paths. If the fuzzer is provided with a corpus containing representative samples of inputs that can be handled by the program being tested, then it is more likely to mutate into an input that triggers unexpected behaviour. This is because the parser is less likely to outright reject the input and the fuzzer more likely to introduce new data to the input that triggers unexpected behaviour while processing the parsed input. Also, it means the fuzzer has less work to do to "discover" new functionality or features.

**Comment by konstin** [comment]
Thanks for the explanation! In this case cpython plus our own test fixture should be a great start

**Comment by addisoncrump** [comment]
In that case, once this lands, I'll update the (re)init-fuzzer script to use cpython source as part of the corpus init.

**Comment by charliermarsh** [comment]
@konstin - Is this ready for review? Tag me when ready :)

**Comment by konstin** [comment]
it is :)

File: crates/ruff_dev/src/main.rs [hunk 1]
```diff
@@ -48,22 +51,29 @@ enum Command {
...
 }
 
-fn main() -> Result<()> {
+fn main() -> Result<ExitCode> {
     let args = Args::parse();
     #[allow(clippy::print_stdout)]
-    match &args.command {
-        Command::GenerateAll(args) => generate_all::main(args)?,
-        Command::GenerateJSONSchema(args) => generate_json_schema::main(args)?,
+    match args.command {
```
Comment (konstin) [inline crates/ruff_dev/src/main.rs:62]: i've changed this back and forth like three times :sweat_smile: i can also roll this back once again

**Review by charliermarsh — APPROVED** [review]

**Pull request merged**